### PR TITLE
Added Salesforce source connector properties file

### PR DESCRIPTION
**Pull request by Nayananga**
Hi, when I was testing this example I get this error,
>[2020-05-03 22:25:15,580] ERROR An exception has occurred before CamelContext startup has finished (org.apache.camel.kafkaconnector.utils.CamelMainSupport:220)
java.lang.IllegalArgumentException: Error configuring property: camel.component.salesforce.password because cannot find component with name salesforce. Make sure you have the component on the classpath
	at org.apache.camel.main.BaseMainSupport.lambda$autoConfigurationFromProperties$15(BaseMainSupport.java:959)
	at org.apache.camel.main.BaseMainSupport.computeProperties(BaseMainSupport.java:1148)
	at org.apache.camel.main.BaseMainSupport.autoConfigurationFromProperties(BaseMainSupport.java:956)
	at org.apache.camel.main.BaseMainSupport.autoconfigure(BaseMainSupport.java:545)
	at org.apache.camel.main.BaseMainSupport.postProcessCamelContext(BaseMainSupport.java:587)
	at org.apache.camel.main.BaseMainSupport.initCamelContext(BaseMainSupport.java:423)
	at org.apache.camel.main.Main.doInit(Main.java:109)
	at org.apache.camel.support.service.ServiceSupport.init(ServiceSupport.java:83)
	at org.apache.camel.support.service.ServiceSupport.start(ServiceSupport.java:112)
	at org.apache.camel.main.MainSupport.run(MainSupport.java:82)
	at org.apache.camel.kafkaconnector.utils.CamelMainSupport$CamelContextStarter.run(CamelMainSupport.java:218)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

and 
>[2020-05-03 22:25:15,589] ERROR WorkerSourceTask{id=CamelSalesforceSourceConnector-0} Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:187)

is this the correct approach? Thanks :)

### CODE REVIEW

File: examples/CamelSalesforceSourceConnector.properties [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name=CamelSalesforceSourceConnector
+connector.class=org.apache.camel.kafkaconnector.CamelSourceConnector
```
Comment (oscerd) [inline examples/CamelSalesforceSourceConnector.properties:19]: You need to use the source connector class for camel-salesforce-kafka-connector jar

**Comment by Nayananga** [comment]
@oscerd Thank you for your suggestion, Now I'm getting a new error, I understand the error but I'm having a hard time finding a fix for it. Any idea?
> Caused by: org.apache.kafka.connect.errors.ConnectException: Failed to find any class that implements Connector and which name matches org.apache.camel.kafkaconnector.salesforce.CamelSalesforceSourceConnector, available connectors are: PluginDesc{klass=class org.apache.camel.kafkaconnector.CamelSinkConnector, name='org.apache.camel.kafkaconnector.CamelSinkConnector', version='0.2.0-SNAPSHOT', encodedVersion=0.2.0-SNAPSHOT, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class org.apache.camel.kafkaconnector.CamelSourceConnector, name='org.apache.camel.kafkaconnector.CamelSourceConnector', version='0.2.0-SNAPSHOT', encodedVersion=0.2.0-SNAPSHOT, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.file.FileStreamSinkConnector, name='org.apache.kafka.connect.file.FileStreamSinkConnector', version='2.5.0', encodedVersion=2.5.0, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.file.FileStreamSourceConnector, name='org.apache.kafka.connect.file.FileStreamSourceConnector', version='2.5.0', encodedVersion=2.5.0, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorCheckpointConnector, name='org.apache.kafka.connect.mirror.MirrorCheckpointConnector', version='1', encodedVersion=1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorHeartbeatConnector, name='org.apache.kafka.connect.mirror.MirrorHeartbeatConnector', version='1', encodedVersion=1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.mirror.MirrorSourceConnector, name='org.apache.kafka.connect.mirror.MirrorSourceConnector', version='1', encodedVersion=1, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockConnector, name='org.apache.kafka.connect.tools.MockConnector', version='2.5.0', encodedVersion=2.5.0, type=connector, typeName='connector', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockSinkConnector, name='org.apache.kafka.connect.tools.MockSinkConnector', version='2.5.0', encodedVersion=2.5.0, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockSourceConnector, name='org.apache.kafka.connect.tools.MockSourceConnector', version='2.5.0', encodedVersion=2.5.0, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.SchemaSourceConnector, name='org.apache.kafka.connect.tools.SchemaSourceConnector', version='2.5.0', encodedVersion=2.5.0, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSinkConnector, name='org.apache.kafka.connect.tools.VerifiableSinkConnector', version='2.5.0', encodedVersion=2.5.0, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSourceConnector, name='org.apache.kafka.connect.tools.VerifiableSourceConnector', version='2.5.0', encodedVersion=2.5.0, type=source, typeName='source', location='classpath'}

File: examples/CamelSalesforceSourceConnector.properties [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name=CamelSalesforceSourceConnector
+connector.class=org.apache.camel.kafkaconnector.salesforce.CamelSalesforceSourceConnector
+key.converter=org.apache.kafka.connect.storage.StringConverter
+value.converter=org.apache.kafka.connect.storage.StringConverter
+
+camel.source.kafka.topic=mytopic
```
Comment (oscerd) [inline examples/CamelSalesforceSourceConnector.properties:23]: this topics and not camel.source.kafka.topic from a couple of days

File: examples/CamelSalesforceSourceConnector.properties [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+camel.source.url=salesforce:CamelKafkaConnectorTestTopic?
```
Comment (oscerd) [inline examples/CamelSalesforceSourceConnector.properties:25]: If you choose to use camel.source.url, you need to specify everything there and avoid camel.source.endpoint.* properties. If you choose to use camel.source.endpoint.* you must avoid camel.source.url

File: examples/CamelSalesforceSourceConnector.properties (hunk 2 above)
Comment (Nayananga) [reply examples/CamelSalesforceSourceConnector.properties:23]: > this topics and not camel.source.kafka.topic from a couple of days

Sorry I didn't understand what you mean by, you meant now this ``` camel.source.kafka.topic``` should change into ``` topics``` ?

Comment (oscerd) [reply examples/CamelSalesforceSourceConnector.properties:23]: Yes. For 0.2.0-SNAPSHOT is topics

**Comment by Nayananga** [comment]
> Can you please show how you're running the example step by step? I think there is something wrong in your classpath

### Here my steps to reproduce,
#### 1) Download the Kafka 2.5.0 release and un-tar it.
#### 2) Add ```export KAFKA_HOME=<your kafka install dir>``` into .bash_profile file.
#### 3) Each time call ```source ~/.bash_profile```
##### 3.1) ```$KAFKA_HOME/bin/zookeeper-server-start.sh $KAFKA_HOME/config/zookeeper.properties```
##### 3.2) ```$KAFKA_HOME/bin/kafka-server-start.sh $KAFKA_HOME/config/server.properties```
##### 3.4)
```
$KAFKA_HOME/bin/kafka-topics.sh --create \
  --zookeeper localhost:2181 \
  --replication-factor 1 \
  --partitions 1 \
  --topic myopic
```
##### 3.5)```$KAFKA_HOME/bin/kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic mytopic --from-beginning```
##### 3.6) ```$KAFKA_HOME/bin/kafka-console-producer.sh --broker-list localhost:9092 --topic mytopic```
#### 4) Add ```export CLASSPATH="$(find core/target/ -type f -name '*.jar'| grep '\-package' | tr '\n' ':')"``` into .bash_profile file
#### 5) clone this projec
#### 6) in project root run ```./mvnw package```
#### 7) call ```source ~/.bash_profile```
Screenshot from 2020-05-15 00-22-49
#####
 7.1) ```$KAFKA_HOME/bin/connect-standalone.sh $KAFKA_HOME/config/connect-standalone.properties config/CamelSourceConnector.properties```

**Comment by Nayananga** [comment]
> > Can you please show how you're running the example step by step? I think there is something wrong in your classpath
> 
> ### Here my steps to reproduce,
> #### 1) Download the Kafka 2.5.0 release and un-tar it.
> #### 2) Add `export KAFKA_HOME=<your kafka install dir>` into .bash_profile file.
> #### 3) Each new window call `source ~/.bash_profile` and then run
> ##### 3.1) `$KAFKA_HOME/bin/zookeeper-server-start.sh $KAFKA_HOME/config/zookeeper.properties`
> ##### 3.2) `$KAFKA_HOME/bin/kafka-server-start.sh $KAFKA_HOME/config/server.properties`
> ##### 3.4)
> 
> ```
> $KAFKA_HOME/bin/kafka-topics.sh --create \
>   --zookeeper localhost:2181 \
>   --replication-factor 1 \
>   --partitions 1 \
>   --topic myopic
> ```
> 
> ##### 3.5)`$KAFKA_HOME/bin/kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic mytopic --from-beginning`
> ##### 3.6) `$KAFKA_HOME/bin/kafka-console-producer.sh --broker-list localhost:9092 --topic mytopic`
> #### 4) Add `export CLASSPATH="$(find core/target/ -type f -name '*.jar'| grep '\-package' | tr '\n' ':')"` into .bash_profile file
> #### 5) clone this projec
> #### 6) in project root run `./mvnw package`
> #### 7) call `source ~/.bash_profile`
> 
> Screenshot from 2020-05-15 00-22-49
> ##### 7.1) `$KAFKA_HOME/bin/connect-standalone.sh $KAFKA_HOME/config/connect-standalone.properties examples/CamelSalesforceSourceConnector.properties`

I followed these steps according to the official documentation, First two examples Simple logger (sink) and Timer (source) working fine, then I tried this salesforce one

**Comment by oscerd** [comment]
I think you're not exporting the correct stuff. The camel Salesforce connector is not under core/target. Please report the printed classpath and check if the Salesforce connector is listed. I don't think so.

**Comment by oscerd** [comment]
You can also have a look at the camel-kafka-connector-examples repo and use the plug-in.path approach used there. There 3 examples already.

**Comment by Nayananga** [comment]
Hi @oscerd, I was able to go through above error after following guide in camel-kafka-connector-examples repository. Now I encountered with this error. Any suggestions? 
```
[2020-05-26 03:20:15,156] ERROR Error starting CamelContext (camel-1) due to exception thrown: Failed to start route route1 because of null (org.apache.camel.impl.engine.AbstractCamelContext:2651)
org.apache.camel.FailedToStartRouteException: Failed to start route route1 because of null
        at org.apache.camel.impl.engine.RouteService.warmUp(RouteService.java:125)
        at org.apache.camel.impl.engine.AbstractCamelContext.doWarmUpRoutes(AbstractCamelContext.java:3313)
        at org.apache.camel.impl.engine.AbstractCamelContext.safelyStartRouteServices(AbstractCamelContext.java:3206)
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartOrResumeRoutes(AbstractCamelContext.java:2992)
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartCamel(AbstractCamelContext.java:2790)
        at org.apache.camel.impl.engine.AbstractCamelContext.doStartContext(AbstractCamelContext.java:2643)
        at org.apache.camel.impl.engine.AbstractCamelContext.doStart(AbstractCamelContext.java:2606)
        at org.apache.camel.support.service.BaseService.start(BaseService.java:115)
        at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2448)
        at org.apache.camel.main.Main.doStart(Main.java:118)
        at org.apache.camel.support.service.BaseService.start(BaseService.java:115)
        at org.apache.camel.main.MainSupport.run(MainSupport.java:83)
        at org.apache.camel.kafkaconnector.utils.CamelMainSupport$CamelContextStarter.run(CamelMainSupport.java:224)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalArgumentException: SObject Class not found for Merchandise__c
        at org.apache.camel.component.salesforce.SalesforceConsumer.<init>(SalesforceConsumer.java:109)
        at org.apache.camel.component.salesforce.SalesforceEndpoint.createConsumer(SalesforceEndpoint.java:91)
        at org.apache.camel.impl.engine.DefaultRoute.addServices(DefaultRoute.java:560)
        at org.apache.camel.impl.engine.DefaultRoute.onStartingServices(DefaultRoute.java:166)
        at org.apache.camel.impl.engine.RouteService.doWarmUp(RouteService.java:153)
        at org.apache.camel.impl.engine.RouteService.warmUp(RouteService.java:123)
        ... 15 more
```

**Comment by oscerd** [comment]
The merchandise class should have been generated so I think you need to pack the salesforce specific classes in a jar and place them in the folder under plugin.path where unzipped the camel-salesforce connector.

**Comment by oscerd** [comment]
We merged an integration test today. I think this PR could be closed

**Pull request closed without merging**